File: Prision/src/main/java/com/gkzxhn/prison/model/iml/CallUserModel.java
package com.gkzxhn.prison.model.iml;

import com.android.volley.AuthFailureError;
import com.gkzxhn.prison.async.VolleyUtils;
import com.gkzxhn.prison.common.Constants;
import com.gkzxhn.prison.entity.ZijingCall;
import com.gkzxhn.prison.model.ICallUserModel;
import com.gkzxhn.prison.utils.XtHttpUtil;
import com.google.gson.Gson;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by Raleigh.Luo on 17/4/13.
 */

public class CallUserModel extends BaseModel implements ICallUserModel{
    /**
     * 获取免费呼叫次数
     * @param onFinishedListener
     */
    @Override
    public void requestFreeTime(VolleyUtils.OnFinishedListener<JSONObject> onFinishedListener) {
        try {
            String url=String.format("%s/%s/access_times",Constants.REQUEST_FREE_MEETING_TIME,
                    getSharedPreferences().getString(Constants.USER_ACCOUNT,""));
            volleyUtils.get(JSONObject.class, url,REQUEST_TAG,onFinishedListener);
        } catch (AuthFailureError authFailureError) {
            authFailureError.printStackTrace();
        }
    }
    @Override
    public void request(String id, VolleyUtils.OnFinishedListener<JSONObject> onFinishedListener) {
        String url=String.format("%s/%s",Constants.REQUEST_MEETING_DETAIL_URL,id);
        try {
            volleyUtils.get(JSONObject.class,url,REQUEST_TAG,onFinishedListener);
        } catch (AuthFailureError authFailureError) {
            authFailureError.printStackTrace();
        }
    }

    @Override
    public void dial(String account, VolleyUtils.OnFinishedListener<JSONObject> onFinishedListener) {
        try {
            String[] strings =account.split("##");
            String protocol = getSharedPreferences().getString(Constants.PROTOCOL, "h323");
            int rate = getSharedPreferences().getInt(Constants.TERMINAL_RATE, 512);
            JSONObject params = new JSONObject();
            params.put("url",String.format("%s:%s**%s",protocol,strings.length > 0?strings[0]:"",strings.length > 1?strings[1]:""));
            params.put("rate",rate);
            volleyUtils.post(XtHttpUtil.DIAL,params,REQUEST_TAG,onFinishedListener);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: Prision/src/main/java/com/gkzxhn/prison/model/iml/CallUserModel.kt
package com.gkzxhn.prison.model.iml

import android.telecom.Call
import android.util.Log
import com.android.volley.AuthFailureError
import com.android.volley.VolleyError
import com.gkzxhn.prison.common.Constants
import com.gkzxhn.prison.model.ICallUserModel
import com.gkzxhn.prison.utils.XtHttpUtil
import com.gkzxhn.prison.async.VolleyUtils
import com.gkzxhn.prison.utils.Utils
import com.starlight.mobile.android.lib.util.ConvertUtil
import org.json.JSONException

import org.json.JSONObject

/**
 * Created by Raleigh.Luo on 17/4/13.
 */

class CallUserModel : CallZijingModel(), ICallUserModel {


    /**
     * 获取免费呼叫次数
     * @param onFinishedListener
     */
    override fun requestFreeTime(onFinishedListener: VolleyUtils.OnFinishedListener<JSONObject>) {
        try {
            val url = String.format("%s?terminalNumber=%s", Constants.REQUEST_FREE_MEETING_TIME,
                    sharedPreferences.getString(Constants.USER_ACCOUNT, ""))
            volleyUtils[JSONObject::class.java, url, REQUEST_TAG, onFinishedListener]
        } catch (authFailureError: AuthFailureError) {
            authFailureError.printStackTrace()
        }

    }

    /**
     * 获取家属信息
     */
    override fun request(familyId: String, onFinishedListener: VolleyUtils.OnFinishedListener<JSONObject>) {
        val url = String.format("%s?id=%s", Constants.REQUEST_MEETING_DETAIL_URL, familyId)
        try {
            volleyUtils[JSONObject::class.java, url, REQUEST_TAG, onFinishedListener]
        } catch (authFailureError: AuthFailureError) {
            authFailureError.printStackTrace()
        }
    }

    /**
     * 查询会见家属
     */
    override fun requestByMettingId(meetingId: String, onFinishedListener: VolleyUtils.OnFinishedListener<JSONObject>) {
        val url = String.format("%s?meetingId=%s", Constants.REQUEST_MEETING_MEMBERS_URL, meetingId)
        try {
            volleyUtils[JSONObject::class.java, url, REQUEST_TAG, onFinishedListener]
        } catch (authFailureError: AuthFailureError) {
            authFailureError.printStackTrace()
        }
    }
    override fun requestFamily(key: String, onFinishedListener: VolleyUtils.OnFinishedListener<JSONObject>) {
        var searchKey=key
        if(!Utils.isPhoneNumber(key))searchKey=ConvertUtil.urlEncode(key)//中文转码
        val url = String.format("%s?key=%s&page=1&rows=1000&jailId=%s", Constants.REQUEST_FAMILY_BY_KEY, searchKey,
                sharedPreferences.getString(Constants.TERMINAL_JIAL_ID,""))
        try {
            volleyUtils[JSONObject::class.java, url, REQUEST_TAG, onFinishedListener]
        } catch (authFailureError: AuthFailureError) {
            authFailureError.printStackTrace()
        }
    }
}
